src/ResourcesSortedAtoZ.js

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import ResourceItem from "./ResourceItem";
+
+class Resources1 extends React.Component {
+  render() {
+    const { resources, search, } = this.props;
+    let query = search.toLowerCase();
+        let sorted = resources.sort(function(a, b) {
```

Comment:
The sort function is the only difference between this component and `Resources.js`. Can you think of a way to extract that function so that, rather than the parent choosing which of two components to render, the parent renders a single component but chooses *which sort function to use*? This would make it much easier to scale to other sorting (e.g. age, if we started storing that) without writing one component per sort order.

Reply:
I have tried it by using one component for both sorting functions but it breaks other code.but I will try to make it better.

Reply:
I'd be interested to see what you tried and what exactly the problem was - maybe push it to a branch and let me know what exact errors you were getting? It seems like just passing in a sort function as a prop would be a simple way to reduce duplication, but I haven't tried it.

Reply:
where exactly should I pass the sort function as a prop?

Reply:
The simplest is probably just to replace the code in `App.js` that picks a *component* (line 71 onwards) with code that picks a *function* instead, then passes that as e.g. the `sortFunction` prop to a single `Resources.js` component.

Reply:
Cool I will try it, Thanks.